### DISCUSSION TITLE
Is it possible to have multiple domains on the same server?

### DISCUSSION THREAD

**Discussion by abishekrsrikaanth** [postalserver/postal#2024] [Help with using Postal] [answered]
So, I want to understand if there was a way to have multiple domains/subdomains configured on postal that is installed on the same server?

**Comment by willpower232** [comment, marked as answer]
When you set up Postal, you add some subdomains to refer to the Postal server itself and then you can add organizations/servers within the web UI to allow you to send from as many domains as you want as they will reference the initial subdomains in their DNS.

**Reply by abishekrsrikaanth** [reply]
Thank you that makes sense. Is there an API to add additional subdomains/domains to Postal?

**Reply by willpower232** [reply]
There is not currently, someone started one but it was closed in favour of something that hasn't arrived yet.

https://github.com/postalserver/postal/pull/992

**Reply by abishekrsrikaanth** [reply]
Are there any plans for this feature in the near future? I want to be able to add domains dynamically for our SaaS application and have them verified by the users without giving them access to the Postal Server. I was hoping this would be out of the box.

**Reply by willpower232** [reply]
Its been a while so I'm not sure what the plan is there. You could try making your own docker image using that branch as a hopefully short term solution.

**Reply by abishekrsrikaanth** [reply]
Thanks, @willpower232. I am new to ruby, is there any docs on how to run the application locally on an M1 Mac to test out the domain API's from the fork? Any help is appreciated.

**Reply by willpower232** [reply]
You should be able to clone the repository, merge the branch and docker build locally. Then you'll probably need the docker-compose.yml file from the install repository to get it actually running and you can follow the normal install.

**Reply by abishekrsrikaanth** [reply]
Thanks that helped. Appreciate your help. Are you available on the discord channel, the postal channel on discord is pretty silent..
